app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/DragAndDropSortInteractionViewModel.kt

```diff
@@ -28,18 +32,25 @@ class DragAndDropSortInteractionViewModel(
   private val allowMultipleItemsInSamePosition: Boolean by lazy {
     interaction.customizationArgsMap["allowMultipleItemsInSamePosition"]?.boolValue ?: false
   }
-  private val choiceStrings: List<String> by lazy {
+  private val choiceSubtitledHtmls: List<SubtitledHtml> by lazy {
     interaction.customizationArgsMap["choices"]
       ?.schemaObjectList
       ?.schemaObjectList
-      ?.map { schemaObject -> schemaObject.subtitledHtml.html }
+      ?.map { schemaObject -> schemaObject.customSchemaValue.subtitledHtml }
       ?: listOf()
   }
 
-  val choiceItems: ArrayList<DragDropInteractionContentViewModel> =
-    computeChoiceItems(choiceStrings, this)
+  private val contentIdHtmlMap: Map<String, String> =
+    choiceSubtitledHtmls.associate { subtitledHtml ->
+      subtitledHtml.contentId to subtitledHtml.html
+    }
+
+  private val _choiceItems: MutableList<DragDropInteractionContentViewModel> =
+    computeChoiceItems(contentIdHtmlMap, choiceSubtitledHtmls, this)
+
+  val choiceItems: List<DragDropInteractionContentViewModel> = _choiceItems
 
-  var isAnswerAvailable = ObservableField<Boolean>(false)
+  private var isAnswerAvailable = ObservableField<Boolean>(false)
```

Comment:
Though this is not part of your PR but lets make it `val`

Reply:
Done. Changes should be available to review soon.